### PR TITLE
Added a way to return a certificate chain

### PR DESCRIPTION
**Pull request by mcjimenez**
Reads all the certificates, in PEM format, from the configured directory and returns the correct certificate chain on SSL requests

### CODE REVIEW

File: src/config.js [hunk 1]
```diff
@@ -40,14 +40,15 @@ exports.consts = {
...
+  ca_files: ['/opt/certs/rootcatid_ROOTCATID_3.pem', '/opt/certs/issuecatid.hi.inet_ISSUECATID_3.pem'],
+  key: '/opt/certs/key_nopwd_pushserv.pem',
+  cert: '/opt/certs/cert_pushserv.pem',
 
   /**
    * Public base URL to receive notifications. This will be the base to
    * append the /notify/12345abcdef… URL
    */
-  publicBaseURL: 'https://localhost:8081/v1',
+  publicBaseURL: 'https://pushdev.srv.openwebdevice.com:8081/v1',
```
Comment (willyaranda) [inline src/config.js]: Please, do not change this file, just add things on the config.js.ic

**Comment by willyaranda** [comment]
@mcjimenez could you put this code on the `src/common/helpers.js` since it will be written just once :)

File: src/config.js [hunk 2]
```diff
@@ -40,14 +40,15 @@ exports.consts = {
    * Self-signed: http://stackoverflow.com/questions/9519707/can-nodejs-generate-ssl-certificates
    */
 
-  key: '../examples/ssl_cert/server-key.pem',
-  cert: '../examples/ssl_cert/server-cert.pem',
+  ca_files: ['/opt/certs/rootcatid_ROOTCATID_3.pem', '/opt/certs/issuecatid.hi.inet_ISSUECATID_3.pem'],
+  key: '/opt/certs/key_nopwd_pushserv.pem',
+  cert: '/opt/certs/cert_pushserv.pem',
```
Comment (frsela) [inline src/config.js:45]: By default I prefer relative paths to be easy deployed on our working machines.
The testing and public certificates should be included into the commit avoiding all developer machines to be setup'ed individually ;)

File: src/config.js (hunk 1 above)
Comment (frsela) [inline src/config.js]: Idem. The default config.js should work on localhost for developer machines

File: src/config.js.ic [hunk 3]
```diff
@@ -39,6 +39,7 @@ exports.consts = {
    * Self-signed: http://stackoverflow.com/questions/9519707/can-nodejs-generate-ssl-certificates
    */
 
+  ca_files: ['/opt/certs/rootcatid_ROOTCATID_3.pem', '/opt/certs/issuecatid.hi.inet_ISSUECATID_3.pem'],
```
Comment (frsela) [inline src/config.js.ic]: For default IC config, ask Marcos to assure this works for them - I suppose they'll update the RPM generation ;)

**Comment by frsela** [comment]
Instead helpers, crypto common lib can be used since this is related to crypto ... WDYT?

File: src/config.js (hunk 2 above)
Comment (AntonioMA) [inline src/config.js:45]: This should be a directory path anyway, not a list of files.

**Comment by AntonioMA** [comment]
It doesn't actually do any crypto... just load a bunch of files from a directory :)

**Comment by frsela** [comment]
Thanks Carmen :)

Can you uplift the patch to develop branch (future next release ;)

File: src/common/helpers.js [hunk 4]
```diff
@@ -71,4 +71,25 @@ function isVersion(version) {
...
\ No newline at end of file
+exports.isVersion = isVersion;
+
+
+function getCaChannel() {
+    
+  var caDir = require('../config.js').consts.caDir;
+  var fs = require('fs');
+
+  var files = fs.readdirSync(consts.caDir);
```
Comment (willyaranda) [inline src/common/helpers.js:82]: @mcjimenez @frsela does this works?

`consts` is not defined, but `caDir` is. if in line 79 you are `var caDir = require('../config.js').consts.caDir;`, you _need_ to use caDir directly, not `consts.caDir`.

File: src/ns_ua/ws_server.js [hunk 5]
```diff
@@ -360,8 +361,9 @@ server.prototype = {
             });
             if (channelsUpdate.length > 0) {
               return cb(channelsUpdate);
+              cb(channelsUpdate);
```
Comment (willyaranda) [inline src/ns_ua/ws_server.js:364]: this line is incorrect, please, fix it in a followup PR

File: src/common/helpers.js [hunk 6]
```diff
@@ -71,4 +71,25 @@ function isVersion(version) {
...
+
+  var i, 
+      len;
+
+  var ca = [];
+
+  for (i = 0, len = files.length; i < len; i++) {
+    ca.push(fs.readFileSync(consts.caDir + "/" + files[i]));
```
Comment (willyaranda) [inline src/common/helpers.js:90]: do we care about Windows? :) http://nodejs.org/api/path.html#path_path_sep

**Pull request merged**